packages/buidler-core/src/internal/cli/analytics.ts

```diff
@@ -0,0 +1,268 @@
+      t: "pageview",
+
+      // Buidler's tracking Id.
+      tid: this._trackingId,
+
+      // Client Id.
+      cid: this._clientId,
+
+      // Document path, must start with a '/'.
+      dp: `/task/${taskName}`,
+
+      // Host name.
+      dh: "buidler.dev",
+
+      // User agent, must be present.
+      // We use it to inform Node version used and OS.
+      // Example:
+      //   Node/v8.12.0 (Darwin 17.7.0)
+      ua: getUserAgent(),
+
+      // We're using the following values (Campaign source, Campaign medium) to track
+      // whether the user is a Developer or CI, as Custom Dimensions are not working for us atm.
+      cs: this._userType,
+      cm: "User Type",
+
+      // We're using custom dimensions for tracking different user projects, and user types (Developer/CI).
+      //
+      // See the following link for docs on these paremeters:
+      // https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#pr_cd_
+      //
+      // See the following link for setting up our custom dimensions in the Google Analytics dashboard
+      // https://support.google.com/tagmanager/answer/6164990
+      //
+      // Custom dimension 1: Project Id
+      // This is computed as the keccak256 hash of the project's absolute path.
+      cd1: this._projectId,
+      // Custom dimension 2: User type
+      //   Possible values: "CI", "Developer".
+      cd2: this._userType,
+      // Custom dimension 3: Buidler Version
+      //   Example: "Buidler 1.0.0-beta.8".
+      cd3: await getBuidlerVersion()
+    };
+  }
+
+  private _sendHit(hit: RawAnalytics): [AbortAnalytics, Promise<void>] {
+    log(`Sending hit for ${hit.dp}`);
+
+    const controller = new AbortController();
+
+    const abortAnalytics = () => {
+      log(`Aborting hit for ${JSON.stringify(hit.dp)}`);
+
+      controller.abort();
+    };
+
+    const hitPayload = qs.stringify(hit);
+
+    log(`Hit payload: ${JSON.stringify(hit)}`);
+
+    const hitPromise = fetch(googleAnalyticsUrl, {
+      body: hitPayload,
+      method: "POST",
+      signal: controller.signal
+    })
+      .then(() => {
+        log(`Hit for ${JSON.stringify(hit.dp)} sent successfully`);
+      })
+      // We're not really interested in handling failed analytics requests
+      .catch(() => {
+        log("Hit request failed");
+      });
+
+    return [abortAnalytics, hitPromise];
+  }
+
+  /**
+   * Checks whether we're using Buidler in development mode (that is, we're working _on_ Buidler).
+   * We don't want the tasks we run at these moments to be tracked, so we disable analytics if so.
+   */
+  private _isLocalDev(): boolean {
+    const executionMode = getExecutionMode();
+
+    return (
+      executionMode === ExecutionMode.EXECUTION_MODE_LINKED ||
+      executionMode === ExecutionMode.EXECUTION_MODE_TS_NODE_TESTS
+    );
+  }
+}
+
+// TODO: Check Windows support for this approach
+const globalBuidlerConfigFile = path.join(
```

Comment:
Maybe move this into `getClientId` or into its own function? I don't see a reason to have it in the global scope, nor compute it when this module is loaded.